src/clients/python/library/tritonclient/utils/shared_memory/shared_memory.cc

```diff
@@ -130,12 +130,21 @@ GetSharedMemoryHandleInfo(
 int
 SharedMemoryRegionDestroy(void* shm_handle)
 {
-  std::string shm_key =
-      reinterpret_cast<SharedMemoryHandle*>(shm_handle)->shm_key_;
+  SharedMemoryHandle* handle =
+      reinterpret_cast<SharedMemoryHandle*>(shm_handle);
+  char* shm_addr = reinterpret_cast<char*>(handle->base_addr_);
+  size_t byte_size = handle->byte_size_;
+  int status = munmap(shm_addr, byte_size);
+  if (status == -1) {
+    return -6;
+  }
+
+  std::string shm_key = handle->shm_key_.c_str();
```

Comment:
You can directly pass the `handle->shm_key_.c_str()` in the shm_unlink function below. No need to create an additional string.

Reply:
Done